### PR TITLE
Refactor: leverage cl-destructuring-bind

### PR DESCRIPTION
**Pull request by progfolio**
Avoid repeated `plist-get` assignments in favor of `cl-destructuring-bind`.

### CODE REVIEW

File: flycheck.el [hunk 1]
```diff
@@ -2095,19 +2095,11 @@ are mandatory.
...
-        (explainer (plist-get properties :error-explainer))
-        (next-checkers (plist-get properties :next-checkers))
-        (file (flycheck-current-load-file))
-        (working-directory (plist-get properties :working-directory)))
-
+  (cl-destructuring-bind (&key enabled explainer interrupt modes next-checkers
+                               predicate print-doc start verify working-directory
+                               ((:error-filter filter) #'identity)
+                               &allow-other-keys)
+      properties
```
Comment (cpitclaudel) [inline flycheck.el:2102]: This looks pretty good, but I think the original idea was to not depend too much on cl-lib.  Is there a `pcase-let` way to do this?

File: flycheck.el [hunk 2]
```diff
@@ -4183,32 +4175,31 @@ The following PROPERTIES constitute an error level:
...
+    (setf (get level 'flycheck-margin-spec)
+          (or margin-spec
+              (flycheck-make-margin-spec
+               flycheck-default-margin-str
+               (or (get level 'flycheck-fringe-face) 'default))))
+    (setf (get level 'flycheck-margin-continuation)
+          (flycheck-make-margin-spec
+           flycheck-default-margin-continuation-str
+           (or (get level 'flycheck-fringe-face) 'default)))
+    (setf (get level 'flycheck-error-list-face) error-list-face)))
```
Comment (cpitclaudel) [inline flycheck.el:4202]: I'm not sure it's a win to separate the put from the get.

File: flycheck.el (hunk 1 above)
Comment (progfolio) [reply flycheck.el:2102]: Not that I'm aware of. I think pcase-let has a similar facility when extended by `map.el` but I do not have that much experience with it and IIRC it's more verbose and not as flexible as `cl-destructuring-bind`.

File: flycheck.el (hunk 2 above)
Comment (progfolio) [reply flycheck.el:4202]: Fair enough

**Pull request closed without merging**